### PR TITLE
Adding autoclaim option

### LINKED ISSUE #25: Perform autoclaim
A very common pattern after registering a sw is to claim the document so next requests are already being handled by the SW.

This can be done with a pretty simple middleware:
```

importScripts('sww.js');

var worker = new ServiceWorkerWare();

// Claim!
worker.use({
  onActivate: function(e) {
    e.waitUntil(self.clients.claim());
  }
});

worker.init();
```
But being so common, sww could do this for the developer by default, or disable it via configuration.

See: https://bugzilla.mozilla.org/show_bug.cgi?id=1175497

### PR DESCRIPTION
**Pull request by lodr**
Fixes #25

### CODE REVIEW

**Comment by lodr** [comment]
Hi @arcturus , can you provide me some feedback here?

File: lib/sww.js [hunk 1]
```diff
@@ -217,17 +223,42 @@ ServiceWorkerWare.normalizeMwAnswer = function (answer, request, response) {
...
-      waitingList.push(mw.onInstall());
+  var installation = this.getFromMiddleware('onInstall');
+  evt.waitUntil(installation);
+};
+
+/**
+ * Walk all the installed middleware asking if they have prerequisites
+ * (on the way of a promise to be resolved) when SW activates.
+ */
+ServiceWorkerWare.prototype.onActivate = function sww_oninstall(evt) {
```
Comment (arcturus) [inline lib/sww.js]: Should this method be called, here:

https://github.com/lodr/serviceworkerware/blob/issue-25-claim/lib/sww.js#L58

Comment (lodr) [inline lib/sww.js]: Sure. I don't know why its not there. I will fix the name for the function as well.

**Comment by lodr** [comment]
@arcturus should `autoClaim` default to `true`?

**Comment by arcturus** [comment]
Yes please, I think it should be the default approach, taking into account our experience working with sw.

@lodr could you rebase and merge?

**Comment by lodr** [comment]
@arcturus do you mind to review the new tests, please?

**Comment by arcturus** [comment]
Changes good look to me, just found two minor things:
- there are some lint errors
- when trying the demo, the `/demo/virtual/xyz` example is not working, perhaps is not related to this patch.

**Comment by arcturus** [comment]
Once fixed the linting, let's merge and then later we can do a  follow up to fix the demo

**Pull request merged**